packages/wrangler/src/__tests__/d1/execute.test.ts

```diff
@@ -50,6 +50,19 @@ describe("execute", () => {
 		).rejects.toThrowError(`Error: can't use --preview with --local`);
 	});
 
+	it("should reject the use of --remote with --local", async () => {
+		setIsTTY(false);
+		writeWranglerToml({
+			d1_databases: [
+				{ binding: "DATABASE", database_name: "db", database_id: "xxxx" },
+			],
+		});
+
+		await expect(
+			runWrangler(`d1 execute db --command "select;" --local --remote`)
+		).rejects.toThrowError(`Error: can't use --remote with --local`);
+	});
+
 	it("should reject the use of --preview with --local with --json", async () => {
```

Comment:
I note that since `--local` is now the default then this test should fail as long as `--remote` is not provided, right?
There is a similar non-json test above.
Should the messaging be changed to something like "must use --remote with --preview"?